compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -142,9 +149,22 @@ public void visit(BLangPackage astPkg) {
 
     public void visit(BLangTypeDefinition astTypeDefinition) {
         Visibility visibility = getVisibility(astTypeDefinition.symbol);
+
+        List<BIRFunction> attachedFuncs;
+        int typeTag = astTypeDefinition.symbol.type.tag;
+        if (typeTag == TypeTags.OBJECT || typeTag == TypeTags.RECORD) { // TODO: records shouldn't have attached funcs
+            attachedFuncs = new ArrayList<>();
+        } else {
+            attachedFuncs = null;
```

Comment:
Can we make this initilization as a default approach, rather than in an else block

Reply:
Also, why do we have to set this to null?

Reply:
I used a script to remove all white space changes since there was lot of unnecessary lines, that have messed things up it seems, will correct.

Reply:
Ya it's bit ugly.

Only object def have attached funcs,
Better to have `Optional<? extends List<? extends BIRFunctions>>`. But this style is not used in Ballerina source, so I used null to represent that.

What would be a better way to do this?

Reply:
@lankavitharana

Reply:
can't we check the size of the list where we have done a null check?

Reply:
+1 I will use an empty list.